_posts/2019-03-23-WassersteinGAN.markdown

```diff
@@ -0,0 +1,528 @@
+
+  **Optional Reading**:
+
+  1. Ch 20 of the DL book, particularly:
+      * Differentiable Generator Networks (Ch 20.10.2)
+          * Description of a broad class of generative models to which GANs belong which will help contextualize GANs when we look at them next week.
+      * Variational Autoencoders (Ch 20.10.3)
+          * Description of another popular class of differentiable generative model which might be nice to contrast to GANs next week.
+      * Evaluating Generative Models (Ch 20.14)
+          * Summary of techniques and challenges for evaluating generative models which might put Theis _et al._’s paper into context.
+
+  **Questions**:
+
+  _The first two questions are here to make sure that you understand what a generative model is and how it differs from a discriminative model. The last two questions are a good barometer for determining your understanding of the challenges involved in training generative models._ 
+
+  1. Fit a [multivariate Gaussian distribution](https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.stats.multivariate_normal.html) to the [Fisher Iris dataset](https://scikit-learn.org/stable/datasets/index.html#iris-plants-dataset) using maximum likelihood estimation (see Section 2.3.4 of PRML for help) then:
+      1. Determine the probability of seeing a flower with a sepal length of 7.9, a sepal width of 4.4, a petal length of 6.9, and a petal width of 2.5.
+      2. Determine the distribution of flowers with a sepal length of 6.3, a sepal width of 4.8, and a petal length of 6.0 (see section 2.3.2 of PRML for help). 
+      3. Generate 20 flower measurements.
+      4. Generate 20 flower measurements with a sepal length of 6.3.
+
+      (congrats you’ve just trained and used a generative model)
+      <details><summary>Solution</summary>
+      <p>
+      <a href="https://github.com/JamesAllingham/DFL-WGAN/blob/master/DFL_WGAN_week2.ipynb">Here</a> is a Jupyter notebook with solutions. Open the notebook on your computer or Google colab to render the characters properly.
+      </p>
+      </details>
+
+  2. Describe in your own words the difference between a generative and a discriminative model.
+      <details><summary>Solution</summary>
+      <p>
+      This is an open ended question but here are some of the differences:
+      <ul>
+        <li>In the generative setting, we usually model \(p(x)\), our models are usually non-deterministic, and we can sample from them.</li>
+        <li>In the discriminative setting, we usually model \(p(y|x)\), our models are often deterministic, and we can't necessarily sample from them.</li>
```

Comment:
I would also add something to the effect of discriminative models typically need labels, generative models don't need them, as their objective is different.

Reply:
Thanks, I added this.